### PR TITLE
Fixes lp#1840470: bootstrap help displays available config keys.

### PR DESCRIPTION
**Pull request by anastasiamac**
## Description of change

Bootstrap command help has been pointing to stale docs site.
Rather than changing the links to point to the newer site that is going to be obsolete soon as well, this PR goes to the source of truth - the code - to list available config keys.

Drive-by: fixes to some variables that conflict with import names.

## QA steps

run 'juju help bootstrap' to see newer help.

## Bug reference

https://bugs.launchpad.net/juju/+bug/1840470

### CODE REVIEW

**Comment by anastasiamac** [comment]
[fix bootstrap help](https://trello.com/c/IyyN4o9G/956-fix-bootstrap-help)

File: cmd/juju/model/configcommand_test.go [hunk 1]
```diff
@@ -38,7 +38,7 @@ func (s *ConfigCommandSuite) TestInit(c *gc.C) {
 			// Test reset
 			desc:       "reset requires arg",
 			args:       []string{"--reset"},
-			errorMatch: "flag needs an argument: --reset",
+			errorMatch: "option needs an argument: --reset",
```
Comment (timClicks) [inline cmd/juju/model/configcommand_test.go]: Nice change. Burn flags burn.

File: cmd/juju/commands/bootstrap.go [hunk 2]
```diff
@@ -204,12 +208,42 @@ type bootstrapCommand struct {
...
+		usageBootstrapDetailsPartTwo))
+
+	return jujucmd.Info(info)
+}
+
+func (c *bootstrapCommand) configDetails() map[string]interface{} {
+	result := map[string]interface{}{}
+	addAll := func(m map[string]interface{}) {
+		for k, v := range m {
+			result[k] = v
```
Comment (timClicks) [inline cmd/juju/commands/bootstrap.go]: Do we want to check that we have overwritten anything? I guess we control the keys so not really a problem.

Comment (anastasiamac) [reply cmd/juju/commands/bootstrap.go]: Not really... Also I believe that controller config would take precedence anyway ; )

**Comment by anastasiamac** [comment]
$$merge$$

**Comment by anastasiamac** [comment]
brand new and unrelated failure \o/

12:15:48 /home/ubuntu/go/bin/dep ensure -vendor-only 
12:16:14 grouped write of manifest, lock and vendor: error while writing out vendor tree: failed to write dep tree: failed to export cloud.google.com/go: 
12:16:14 	(1) failed to list versions for https://code.googlesource.com/gocloud: fatal: unable to access 'https://code.googlesource.com/gocloud/': gnutls_handshake() failed: The TLS connection was non-properly terminated.
12:16:14 : exit status 128
12:16:14 
12:16:14 Makefile:74: recipe for target 'dep' failed
12:16:14 make: *** [dep] Error 1

**Comment by anastasiamac** [comment]
$$merge$$

**Comment by anastasiamac** [comment]
unrelated failure in github.com/juju/juju/apiserver/facades/agent/caasoperator.TestAll

**Comment by anastasiamac** [comment]
$$merge$$

**Pull request merged**